cfme/cloud/keypairs.py

```diff
@@ -208,6 +208,16 @@ def create(self, name, provider, public_key=None, cancel=False):
         view.flash.assert_success_message(flash_message)
         return self.instantiate(name, provider, public_key=public_key)
 
+    def all(self):
+        view = navigate_to(self, 'All')
+        result = []
+        for _ in view.entities.paginator.pages():
+            for name in view.entities.entity_names:
+                # provider is needed to instantiate but keypairs are not tied to the provider
```

Comment:
With what you're pointing out here, I think you should just update the `attr.ib` definition with a default on the Keypair entity:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/cloud/keypairs.py#L123